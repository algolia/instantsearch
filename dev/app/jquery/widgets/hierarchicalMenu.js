/* eslint-disable import/default */
import instantsearch from '../../../../index';

const formatMenuEntry = (createURL, lvl = 0) => item => {
  const countHTML = `
    <span class="pull-right facet-count">
      ${item.count}
    </span>
  `;

  if (
    item.isRefined === true &&
    Array.isArray(item.data) &&
    item.data.length > 0
  ) {
    return `
      <div ${lvl === 0 ? 'class="hierarchical-categories-list"' : ''}>
        <a
          href="${createURL(item.value)}"
          class="facet-value clearfix"
          data-refine-value="${item.value}"
        >
          <strong>${item.label}</strong> ${countHTML}
        </a>
        <div class="hierarchical-categories-list ais-hierarchical-menu--list__lvl${lvl +
          1}">
          ${item.data.map(formatMenuEntry(createURL, lvl + 1)).join('')}
        </div>
      </div>
    `;
  }

  return `
    <div>
      <a
        href="${createURL(item.value)}"
        class="facet-value clearfix"
        data-refine-value="${item.value}"
      >
        ${
          item.isRefined ? `<strong>${item.label}</strong>` : item.label
        } ${countHTML}
      </a>
    </div>
  `;
};

const renderFn = (
  { createURL, items, refine, widgetParams: { containerNode } },
  isFirstRendering
) => {
  if (isFirstRendering) {
    const markup = window.$(`
      <div class="facet-title">Custom hierarchical</div>
      <div id="custom-hierarchical-menu__container"></div>
    `);
    containerNode.append(markup);
  }

  // remove event listeners before replacing markup
  containerNode.find('a[data-refine-value]').each(function() {
    window.$(this).off('click');
  });

  if (items && items.length > 0) {
    // replace markup with items
    const menuItems = items.map(formatMenuEntry(createURL)).join('');

    containerNode.find('#custom-hierarchical-menu__container').html(menuItems);

    // bind links with `data-refine-value`
    containerNode.find('a[data-refine-value]').each(function() {
      window.$(this).on('click', e => {
        e.preventDefault();
        refine(window.$(this).data('refine-value'));
      });
    });
  }
};

export default instantsearch.connectors.connectHierarchicalMenu(renderFn);
