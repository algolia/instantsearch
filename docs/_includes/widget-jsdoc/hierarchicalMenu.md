<h4 class="no-toc">Parameters</h4>
<p class="attr-name">
<span class='attr-required'>`options.container`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>DOMElement</code>)</span>
</p>
<p class="attr-description important">CSS Selector or DOMElement to insert the widget</p>
<p class="attr-name">
<span class='attr-required'>`options.attributes`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description important">Array of attributes to use to generate the hierarchy of the menu. See the example for the convention to follow.</p>
<p class="attr-name">
<span class='attr-optional important'>`options.limit`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">10</code>(<code>number</code>)</span>
</p>
<p class="attr-description important">How much facet values to get</p>
<p class="attr-name">
<span class='attr-optional important'>`options.separator`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">&quot;&gt;&quot;</code>(<code>string</code>)</span>
</p>
<p class="attr-description important">Separator used in the attributes to separate level values.</p>
<p class="attr-name">
<span class='attr-optional'>`options.rootPath`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code>)</span>
</p>
<p class="attr-description">Prefix path to use if the first level is not the root level.</p>
<p class="attr-name">
<span class='attr-optional'>`options.showParentLevel`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">false</code>(<code>string</code>)</span>
</p>
<p class="attr-description">Show the parent level of the current refined value</p>
<p class="attr-name">
<span class='attr-optional'>`options.sortBy`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">[&#x27;name:asc&#x27;]</code>(<code>Array.&lt;string&gt;</code> &#124; <code>function</code>)</span>
</p>
<p class="attr-description">How to sort refinements. Possible values: `count|isRefined|name:asc|name:desc`.   You can also use a sort function that behaves like the standard Javascript [compareFunction](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Syntax).</p>
<p class="attr-name">
<span class='attr-optional'>`options.templates`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>Object</code>)</span>
</p>
<p class="attr-description">Templates to use for the widget</p>
<p class="attr-name">
<span class='attr-optional'>`options.templates.header`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">&#x27;&#x27;</code>(<code>string</code> &#124; <code>function</code>)</span>
</p>
<p class="attr-description">Header template (root level only)</p>
<p class="attr-name">
<span class='attr-optional'>`options.templates.item`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>function</code>)</span>
</p>
<p class="attr-description">Item template, provided with `name`, `count`, `isRefined`, `url` data properties</p>
<p class="attr-name">
<span class='attr-optional'>`options.templates.footer`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">&#x27;&#x27;</code>(<code>string</code> &#124; <code>function</code>)</span>
</p>
<p class="attr-description">Footer template (root level only)</p>
<p class="attr-name">
<span class='attr-optional'>`options.transformData.item`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>function</code>)</span>
</p>
<p class="attr-description">Method to change the object passed to the `item` template</p>
<p class="attr-name">
<span class='attr-optional'>`options.autoHideContainer`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">true</code>(<code>boolean</code>)</span>
</p>
<p class="attr-description">Hide the container when there are no items in the menu</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>Object</code>)</span>
</p>
<p class="attr-description">CSS classes to add to the wrapping elements</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.root`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to the root element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.header`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to the header element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.body`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to the body element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.footer`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to the footer element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.list`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to the list element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.item`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to each item element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.depth`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to each item element to denote its depth. The actual level will be appended to the given class name (ie. if `depth` is given, the widget will add `depth0`, `depth1`, ... according to the level of each item).</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.active`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to each active element</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.link`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to each link (when using the default template)</p>
<p class="attr-name">
<span class='attr-optional'>`options.cssClasses.count`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>string</code> &#124; <code>Array.&lt;string&gt;</code>)</span>
</p>
<p class="attr-description">CSS class to add to each count element (when using the default template)</p>
<p class="attr-name">
<span class='attr-optional'>`options.collapsible`<span class="show-description">…</span></span>
  <span class="attr-infos">Default:<code class="attr-default">false</code>(<code>object</code> &#124; <code>boolean</code>)</span>
</p>
<p class="attr-description">Hide the widget body and footer when clicking on header</p>
<p class="attr-name">
<span class='attr-optional'>`options.collapsible.collapsed`<span class="show-description">…</span></span>
  <span class="attr-infos">(<code>boolean</code>)</span>
</p>
<p class="attr-description">Initial collapsed state of a collapsible widget</p>

<p class="attr-legend">* <span>Required</span></p>
