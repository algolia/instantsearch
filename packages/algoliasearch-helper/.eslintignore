dist/
documentation-src/
documentation/
scripts/rollup.esm.config.js
*.ts
