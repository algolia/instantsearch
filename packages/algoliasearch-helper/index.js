import AlgoliaSearchHelper from './src/algoliasearch.helper.js';
import RecommendParameters from './src/RecommendParameters/index.js';
import RecommendResults from './src/RecommendResults/index.js';
import SearchParameters from './src/SearchParameters/index.js';
import SearchResults from './src/SearchResults/index.js';

/**
 * The algoliasearchHelper module is the function that will let its
 * contains everything needed to use the Algoliasearch
 * Helper. It is a also a function that instanciate the helper.
 * To use the helper, you also need the Algolia JS client v3.
 * @example
 * //using the UMD build
 * var client = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76');
 * var helper = algoliasearchHelper(client, 'bestbuy', {
 *   facets: ['shipping'],
 *   disjunctiveFacets: ['category']
 * });
 * helper.on('result', function(event) {
 *   console.log(event.results);
 * });
 * helper
 *   .toggleFacetRefinement('category', 'Movies & TV Shows')
 *   .toggleFacetRefinement('shipping', 'Free shipping')
 *   .search();
 * @example
 * // The helper is an event emitter using the node API
 * helper.on('result', updateTheResults);
 * helper.once('result', updateTheResults);
 * helper.removeListener('result', updateTheResults);
 * helper.removeAllListeners('result');
 * @module algoliasearchHelper
 * @param  {AlgoliaSearch} client an AlgoliaSearch client
 * @param  {string} index the name of the index to query
 * @param  {SearchParameters|object} opts an object defining the initial config of the search. It doesn't have to be a {SearchParameters}, just an object containing the properties you need from it.
 * @param {SearchResultsOptions|object} searchResultsOptions an object defining the options to use when creating the search results.
 * @return {AlgoliaSearchHelper} The helper instance
 */

function algoliasearchHelper(client, index, opts, searchResultsOptions) {
  return new AlgoliaSearchHelper(client, index, opts, searchResultsOptions);
}

import version from './src/version.js';
algoliasearchHelper.version = version;

/**
 * Constructor for the Helper.
 * @member module:algoliasearchHelper.AlgoliaSearchHelper
 * @type {AlgoliaSearchHelper}
 */
algoliasearchHelper.AlgoliaSearchHelper = AlgoliaSearchHelper;

/**
 * Constructor for the object containing all the parameters of the search.
 * @member module:algoliasearchHelper.SearchParameters
 * @type {SearchParameters}
 */
algoliasearchHelper.SearchParameters = SearchParameters;

/**
 * Constructor for the object containing all the parameters for Recommend.
 * @member module:algoliasearchHelper.RecommendParameters
 * @type {RecommendParameters}
 */
algoliasearchHelper.RecommendParameters = RecommendParameters;

/**
 * Constructor for the object containing the results of the search.
 * @member module:algoliasearchHelper.SearchResults
 * @type {SearchResults}
 */
algoliasearchHelper.SearchResults = SearchResults;

/**
 * Constructor for the object containing the results for Recommend.
 * @member module:algoliasearchHelper.RecommendResults
 * @type {RecommendResults}
 */
algoliasearchHelper.RecommendResults = RecommendResults;

export default algoliasearchHelper;
