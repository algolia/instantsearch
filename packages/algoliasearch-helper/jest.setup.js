/* eslint-env jest */

'use strict';

jest.setTimeout(20000);
