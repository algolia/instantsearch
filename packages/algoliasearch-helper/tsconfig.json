{
  "compilerOptions": {
    "noEmit": true,
    "lib": ["es2015"],
    "esModuleInterop": true
  },
  "files": ["./index.d.ts", "test/types.ts"]
}
