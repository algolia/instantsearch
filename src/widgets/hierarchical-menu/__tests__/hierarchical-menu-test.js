import { SearchParameters } from 'algoliasearch-helper';
import hierarchicalMenu from '../hierarchical-menu';

describe('hierarchicalMenu()', () => {
  let container;
  let attributes;
  let options;
  let widget;
  let ReactDOM;

  beforeEach(() => {
    container = document.createElement('div');
    attributes = ['hello', 'world'];
    options = {};
    ReactDOM = { render: jest.fn() };
    hierarchicalMenu.__Rewire__('render', ReactDOM.render);
  });

  describe('instantiated with wrong parameters', () => {
    it('should fail if no attributes', () => {
      options = { container, attributes: undefined };
      expect(() => hierarchicalMenu(options)).toThrow(/^Usage:/);
    });

    it('should fail if attributes empty', () => {
      options = { container, attributes: [] };
      expect(() => hierarchicalMenu(options)).toThrow(/^Usage:/);
    });

    it('should fail if no container', () => {
      options = { container: undefined, attributes };
      expect(() => hierarchicalMenu(options)).toThrow(/^Usage:/);
    });
  });

  describe('getConfiguration', () => {
    beforeEach(() => {
      options = { container, attributes };
    });

    it('has defaults', () => {
      expect(hierarchicalMenu(options).getConfiguration({})).toEqual({
        hierarchicalFacets: [
          {
            name: 'hello',
            rootPath: null,
            attributes: ['hello', 'world'],
            separator: ' > ',
            showParentLevel: true,
          },
        ],
        maxValuesPerFacet: 10,
      });
    });

    it('understand the separator option', () => {
      expect(
        hierarchicalMenu({ separator: ' ? ', ...options }).getConfiguration({})
      ).toEqual({
        hierarchicalFacets: [
          {
            name: 'hello',
            rootPath: null,
            attributes: ['hello', 'world'],
            separator: ' ? ',
            showParentLevel: true,
          },
        ],
        maxValuesPerFacet: 10,
      });
    });

    it('understand the showParentLevel option', () => {
      expect(
        hierarchicalMenu({
          showParentLevel: false,
          ...options,
        }).getConfiguration({})
      ).toEqual({
        hierarchicalFacets: [
          {
            name: 'hello',
            rootPath: null,
            attributes: ['hello', 'world'],
            separator: ' > ',
            showParentLevel: false,
          },
        ],
        maxValuesPerFacet: 10,
      });
    });

    it('understand the rootPath option', () => {
      expect(
        hierarchicalMenu({ rootPath: 'Beer', ...options }).getConfiguration({})
      ).toEqual({
        hierarchicalFacets: [
          {
            name: 'hello',
            rootPath: 'Beer',
            attributes: ['hello', 'world'],
            separator: ' > ',
            showParentLevel: true,
          },
        ],
        maxValuesPerFacet: 10,
      });
    });

    describe('limit option', () => {
      it('configures maxValuesPerFacet', () =>
        expect(
          hierarchicalMenu({ limit: 20, ...options }).getConfiguration({})
            .maxValuesPerFacet
        ).toBe(20));

      it('uses provided maxValuesPerFacet when higher', () =>
        expect(
          hierarchicalMenu({ limit: 20, ...options }).getConfiguration({
            maxValuesPerFacet: 30,
          }).maxValuesPerFacet
        ).toBe(30));

      it('ignores provided maxValuesPerFacet when lower', () =>
        expect(
          hierarchicalMenu({ limit: 10, ...options }).getConfiguration({
            maxValuesPerFacet: 3,
          }).maxValuesPerFacet
        ).toBe(10));
    });
  });

  describe('render', () => {
    let results;
    let data;
    let helper;
    let state;
    let createURL;

    beforeEach(() => {
      data = { data: [{ name: 'foo' }, { name: 'bar' }] };
      results = { getFacetValues: jest.fn(() => data) };
      helper = {
        toggleRefinement: jest.fn().mockReturnThis(),
        search: jest.fn(),
      };
      state = new SearchParameters();
      state.toggleRefinement = jest.fn();
      options = { container, attributes };
      createURL = () => '#';
    });

    it('understand provided cssClasses', () => {
      const userCssClasses = {
        root: 'root',
        noRefinementRoot: 'noRefinementRoot',
        searchBox: 'searchBox',
        list: 'list',
        childList: 'childList',
        item: 'item',
        selectedItem: 'selectedItem',
        parentItem: 'parentItem',
        link: 'link',
        label: 'label',
        count: 'count',
        noResults: 'noResults',
        showMore: 'showMore',
        disabledShowMore: 'disabledShowMore',
      };

      widget = hierarchicalMenu({ ...options, cssClasses: userCssClasses });
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      expect(ReactDOM.render.mock.calls[0][0]).toMatchSnapshot();
    });

    it('calls ReactDOM.render', () => {
      widget = hierarchicalMenu(options);
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      expect(ReactDOM.render).toHaveBeenCalledTimes(1);
      expect(ReactDOM.render.mock.calls[0][0]).toMatchSnapshot();
    });

    it('asks for results.getFacetValues', () => {
      widget = hierarchicalMenu(options);
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      expect(results.getFacetValues).toHaveBeenCalledTimes(1);
      expect(results.getFacetValues).toHaveBeenCalledWith('hello', {
        sortBy: ['name:asc'],
      });
    });

    it('has a sortBy option', () => {
      widget = hierarchicalMenu({ ...options, sortBy: ['count:asc'] });
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      expect(results.getFacetValues).toHaveBeenCalledTimes(1);
      expect(results.getFacetValues).toHaveBeenCalledWith('hello', {
        sortBy: ['count:asc'],
      });
    });

    it('has a templates option', () => {
      widget = hierarchicalMenu({
        ...options,
        templates: {
          item: 'item2',
        },
      });
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      expect(ReactDOM.render.mock.calls[0][0]).toMatchSnapshot();
    });

    it('has a transformItems options', () => {
      widget = hierarchicalMenu({
        ...options,
        transformItems: items =>
          items.map(item => ({ ...item, transformed: true })),
      });

      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });

      expect(ReactDOM.render.mock.calls[0][0]).toMatchSnapshot();
    });

    it('sets facetValues to empty array when no results', () => {
      data = {};
      widget = hierarchicalMenu(options);
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      expect(ReactDOM.render.mock.calls[0][0]).toMatchSnapshot();
    });

    it('has a toggleRefinement method', () => {
      widget = hierarchicalMenu(options);
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      const elementToggleRefinement =
        ReactDOM.render.mock.calls[0][0].props.toggleRefinement;
      elementToggleRefinement('mom');
      expect(helper.toggleRefinement).toHaveBeenCalledTimes(1);
      expect(helper.toggleRefinement).toHaveBeenCalledWith('hello', 'mom');
      expect(helper.search).toHaveBeenCalledTimes(1);
    });

    it('has a limit option', () => {
      const secondLevel = [
        { name: 'six', path: 'six' },
        { name: 'seven', path: 'seven' },
        { name: 'eight', path: 'eight' },
        { name: 'nine', path: 'nine' },
      ];

      const firstLevel = [
        { name: 'one', path: 'one' },
        { name: 'two', path: 'two', data: secondLevel },
        { name: 'three', path: 'three' },
        { name: 'four', path: 'four' },
        { name: 'five', path: 'five' },
      ];

      data = { data: firstLevel };
      const expectedFacetValues = [
        { label: 'one', value: 'one' },
        {
          label: 'two',
          value: 'two',
          data: [
            { label: 'six', value: 'six' },
            { label: 'seven', value: 'seven' },
            { label: 'eight', value: 'eight' },
          ],
        },
        { label: 'three', value: 'three' },
      ];
      widget = hierarchicalMenu({ ...options, limit: 3 });
      widget.init({ helper, createURL, instantSearchInstance: {} });
      widget.render({ results, state });
      const actualFacetValues =
        ReactDOM.render.mock.calls[0][0].props.facetValues;
      expect(actualFacetValues).toEqual(expectedFacetValues);
    });
  });

  afterEach(() => {
    hierarchicalMenu.__ResetDependency__('render');
  });
});
