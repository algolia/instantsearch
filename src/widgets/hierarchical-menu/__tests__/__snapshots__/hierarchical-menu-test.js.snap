// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hierarchicalMenu() render calls ReactDOM.render 1`] = `
<HeaderFooter-AutoHide
  collapsible={false}
  createURL={[Function]}
  cssClasses={
    Object {
      "active": "ais-hierarchical-menu--item__active",
      "body": "ais-hierarchical-menu--body",
      "count": "ais-hierarchical-menu--count",
      "depth": "ais-hierarchical-menu--list__lvl",
      "footer": "ais-hierarchical-menu--footer",
      "header": "ais-hierarchical-menu--header",
      "item": "ais-hierarchical-menu--item",
      "link": "ais-hierarchical-menu--link",
      "list": "ais-hierarchical-menu--list",
      "root": "ais-hierarchical-menu",
    }
  }
  facetValues={
    Array [
      Object {
        "label": "foo",
        "value": undefined,
      },
      Object {
        "label": "bar",
        "value": undefined,
      },
    ]
  }
  shouldAutoHideContainer={false}
  templateProps={
    Object {
      "templates": Object {
        "footer": "",
        "header": "",
        "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\">{{label}} <span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      },
      "templatesConfig": undefined,
      "transformData": undefined,
      "useCustomCompileOptions": Object {
        "footer": false,
        "header": false,
        "item": false,
      },
    }
  }
  toggleRefinement={[Function]}
/>
`;

exports[`hierarchicalMenu() render has a templates option 1`] = `
<HeaderFooter-AutoHide
  collapsible={false}
  createURL={[Function]}
  cssClasses={
    Object {
      "active": "ais-hierarchical-menu--item__active",
      "body": "ais-hierarchical-menu--body",
      "count": "ais-hierarchical-menu--count",
      "depth": "ais-hierarchical-menu--list__lvl",
      "footer": "ais-hierarchical-menu--footer",
      "header": "ais-hierarchical-menu--header",
      "item": "ais-hierarchical-menu--item",
      "link": "ais-hierarchical-menu--link",
      "list": "ais-hierarchical-menu--list",
      "root": "ais-hierarchical-menu",
    }
  }
  facetValues={
    Array [
      Object {
        "label": "foo",
        "value": undefined,
      },
      Object {
        "label": "bar",
        "value": undefined,
      },
    ]
  }
  shouldAutoHideContainer={false}
  templateProps={
    Object {
      "templates": Object {
        "footer": "footer2",
        "header": "header2",
        "item": "item2",
      },
      "templatesConfig": undefined,
      "transformData": undefined,
      "useCustomCompileOptions": Object {
        "footer": true,
        "header": true,
        "item": true,
      },
    }
  }
  toggleRefinement={[Function]}
/>
`;

exports[`hierarchicalMenu() render has a transformItems options 1`] = `
<HeaderFooter-AutoHide
  collapsible={false}
  createURL={[Function]}
  cssClasses={
    Object {
      "active": "ais-hierarchical-menu--item__active",
      "body": "ais-hierarchical-menu--body",
      "count": "ais-hierarchical-menu--count",
      "depth": "ais-hierarchical-menu--list__lvl",
      "footer": "ais-hierarchical-menu--footer",
      "header": "ais-hierarchical-menu--header",
      "item": "ais-hierarchical-menu--item",
      "link": "ais-hierarchical-menu--link",
      "list": "ais-hierarchical-menu--list",
      "root": "ais-hierarchical-menu",
    }
  }
  facetValues={
    Array [
      Object {
        "label": "foo",
        "transformed": true,
        "value": undefined,
      },
      Object {
        "label": "bar",
        "transformed": true,
        "value": undefined,
      },
    ]
  }
  shouldAutoHideContainer={false}
  templateProps={
    Object {
      "templates": Object {
        "footer": "",
        "header": "",
        "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\">{{label}} <span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      },
      "templatesConfig": undefined,
      "transformData": undefined,
      "useCustomCompileOptions": Object {
        "footer": false,
        "header": false,
        "item": false,
      },
    }
  }
  toggleRefinement={[Function]}
/>
`;

exports[`hierarchicalMenu() render sets facetValues to empty array when no results 1`] = `
<HeaderFooter-AutoHide
  collapsible={false}
  createURL={[Function]}
  cssClasses={
    Object {
      "active": "ais-hierarchical-menu--item__active",
      "body": "ais-hierarchical-menu--body",
      "count": "ais-hierarchical-menu--count",
      "depth": "ais-hierarchical-menu--list__lvl",
      "footer": "ais-hierarchical-menu--footer",
      "header": "ais-hierarchical-menu--header",
      "item": "ais-hierarchical-menu--item",
      "link": "ais-hierarchical-menu--link",
      "list": "ais-hierarchical-menu--list",
      "root": "ais-hierarchical-menu",
    }
  }
  facetValues={Array []}
  shouldAutoHideContainer={true}
  templateProps={
    Object {
      "templates": Object {
        "footer": "",
        "header": "",
        "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\">{{label}} <span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      },
      "templatesConfig": undefined,
      "transformData": undefined,
      "useCustomCompileOptions": Object {
        "footer": false,
        "header": false,
        "item": false,
      },
    }
  }
  toggleRefinement={[Function]}
/>
`;

exports[`hierarchicalMenu() render sets shouldAutoHideContainer to true when no results 1`] = `
<HeaderFooter-AutoHide
  collapsible={false}
  createURL={[Function]}
  cssClasses={
    Object {
      "active": "ais-hierarchical-menu--item__active",
      "body": "ais-hierarchical-menu--body",
      "count": "ais-hierarchical-menu--count",
      "depth": "ais-hierarchical-menu--list__lvl",
      "footer": "ais-hierarchical-menu--footer",
      "header": "ais-hierarchical-menu--header",
      "item": "ais-hierarchical-menu--item",
      "link": "ais-hierarchical-menu--link",
      "list": "ais-hierarchical-menu--list",
      "root": "ais-hierarchical-menu",
    }
  }
  facetValues={Array []}
  shouldAutoHideContainer={true}
  templateProps={
    Object {
      "templates": Object {
        "footer": "",
        "header": "",
        "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\">{{label}} <span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      },
      "templatesConfig": undefined,
      "transformData": undefined,
      "useCustomCompileOptions": Object {
        "footer": false,
        "header": false,
        "item": false,
      },
    }
  }
  toggleRefinement={[Function]}
/>
`;

exports[`hierarchicalMenu() render understand provided cssClasses 1`] = `
<HeaderFooter-AutoHide
  collapsible={false}
  createURL={[Function]}
  cssClasses={
    Object {
      "active": "ais-hierarchical-menu--item__active active",
      "body": "ais-hierarchical-menu--body body",
      "count": "ais-hierarchical-menu--count count",
      "depth": "ais-hierarchical-menu--list__lvl",
      "footer": "ais-hierarchical-menu--footer footer",
      "header": "ais-hierarchical-menu--header header",
      "item": "ais-hierarchical-menu--item item",
      "link": "ais-hierarchical-menu--link link",
      "list": "ais-hierarchical-menu--list list",
      "root": "ais-hierarchical-menu root cx",
    }
  }
  facetValues={
    Array [
      Object {
        "label": "foo",
        "value": undefined,
      },
      Object {
        "label": "bar",
        "value": undefined,
      },
    ]
  }
  shouldAutoHideContainer={false}
  templateProps={
    Object {
      "templates": Object {
        "footer": "",
        "header": "",
        "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\">{{label}} <span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      },
      "templatesConfig": undefined,
      "transformData": undefined,
      "useCustomCompileOptions": Object {
        "footer": false,
        "header": false,
        "item": false,
      },
    }
  }
  toggleRefinement={[Function]}
/>
`;
